### ISSUE TITLE
:cljc :features config seems to be suppressing linter

### ISSUE THREAD

**Issue by lread** [clj-kondo/clj-kondo#2821]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**
v2026.04.15

**platform**
native on linux

**problem**
On babashka fs, we noticed that clj-kondo was not reporting a redundant nested str call (https://github.com/babashka/fs/pull/258#issuecomment-4338208587). The suppression seems to be related to the usage of `:cljc` `:features` config.

**repro**
Given `fiddle.cljc`
```clojure
(ns fiddle)

;; should emit redundant-nested-call
(str "foo" (str 1))

;; should emit type-mismatch
(+ 1 "foo")
```

✔️ Linting without `:cljc` config works as expected:
```
$ clj-kondo --lint fiddle.cljc
fiddle.cljc:4:12: info: Redundant nested call: str
fiddle.cljc:7:6: error: Expected: number, received: string.
linting took 15ms, errors: 1, warnings: 0
```

But if we include `:cljc` config, notice that the redundant nested call linter finding is not reported.

❌ Not reported for `{:cljc {:features [:cljs]}}`
```
$ clj-kondo --lint fiddle.cljc --config '{:cljc {:features [:cljs]}}'
fiddle.cljc:7:6: error: Expected: number, received: string.
linting took 10ms, errors: 1, warnings: 0
```

❌ Not reported for `{:cljc {:features [:clj]}}`
```
$ clj-kondo --lint fiddle.cljc --config '{:cljc {:features [:clj]}}'     
fiddle.cljc:7:6: error: Expected: number, received: string.
linting took 10ms, errors: 1, warnings: 0
```

✔️ Works for `{:cljc {:features [:clj :cljs]}}`
```
clj-kondo --lint fiddle.cljc --config '{:cljc {:features [:clj :cljs]}}'
fiddle.cljc:4:12: info: Redundant nested call: str
fiddle.cljc:7:6: error: Expected: number, received: string.
linting took 17ms, errors: 1, warnings: 0
```

**expected behavior**
I would expect that the redundant nested call linter not be suppressed for any of the above examples.

**diagnosis**
I've not done any, sorry. 
I'm guessing this is not limited to the `:redundant-nested-call` linter, but I suppose it could be.

**next steps**
I could take a peek if you like.

**Comment by borkdude** [comment]
A peek is surely welcome!

**Comment by lread** [comment]
I think I've narrowed it down to `filter-findings`:
https://github.com/clj-kondo/clj-kondo/blob/6a54fa40373d169372335e6f13f014148883ff9d/src/clj_kondo/impl/core.clj#L714-L751

I'm not yet understanding the need for special checks on specific linters, but think that these linters get filtered out when config specifies a single `:cljc` `:features`.

We see the same symptom for the redundant do linter:

✔️ both `:clj` and `:cljs` :features:
```
$ clj-kondo --lang cljc --lint - <<< '(do (+ 1 2))'
<stdin>:1:1: warning: redundant do
linting took 16ms, errors: 0, warnings: 1

$ clj-kondo --lang cljc --config '{:cljc {:features [:clj :cljs]}}' --lint - <<< '(do (+ 1 2))'
<stdin>:1:1: warning: redundant do
linting took 16ms, errors: 0, warnings: 1
```

❌ only one of `:clj` or `:cljs` :features:
```
$ clj-kondo --lang cljc --config '{:cljc {:features [:clj]}}' --lint - <<< '(do (+ 1 2))'                  
linting took 14ms, errors: 0, warnings: 0

$ clj-kondo --lang cljc --config '{:cljc {:features [:cljs]}}' --lint - <<< '(do (+ 1 2))'
linting took 12ms, errors: 0, warnings: 0
```

**Comment by borkdude** [comment]
> I'm not yet understanding the need for special checks on specific linters

The idea here is to only warn when both cljc features have the warning. If not, then there's probably some reader conditional in there which makes fixing the warning very awkward and not worth it.

We should probably count the features in the config to fix this.

**Comment by lread** [comment]
Ah! Thanks, that helps.
